src/content/blog/2020-12-07-roadmap-2021.md

```diff
@@ -0,0 +1,239 @@
+
+I could think of a special mode, which do no apply chunking and instead emit the raw (processed) modules connected via ESM imports and exports (or also CommonJS `require`).
+So this basically means loaders, module graph and asset optimizations run, but no chunk graph is created and each module in the module graph is emitted as separate file.
+
+#### Strict mode warnings
+
+When generating a ESM bundle, all contained code will be forced to strict mode.
+For many modules this isn't an issue, but there are a few older packages that might have trouble with the different semantic.
+We want to show warnings for these cases.
+
+### More first-class citizen
+
+webpack 4 and 5 did a lot work to support non-JS module types, and webpack 5 already supports some module types by default: JS (ESM/CJS/AMD), JSON, WebAssembly, Asset.
+Since webpack 5 one of our long term goals it to become a web-app optimizer, with the goal of supporting everything the browser supports.
+So technically a vanilla web app should work out-of-the-box with webpack, but being optimized on the go.
+
+The initial webpack 5 release already did some major steps in this direction: `new Worker` is supported natively. `new URL(...)` is supported natively (assets).
+
+WebAssembly and JSON is already supported, even while the proposals are not finished yet.
+
+But two resource types are still missing for the complete story: HTML and CSS.
+
+#### CSS as modules
+
+Currently webpack supports CSS via `css-loader`, `style-loader` or `mini-css-extract-plugin`.
+This is working pretty well, but I think we can do more by supporting CSS as native module type in webpack.
+
+A major benefit would be the developer experience: The `mini-css-extract-plugin` configuration is not the easiest and getting rid of it would simplify at lot for the developer.
+Of course that doesn't mean you can add additional customization on top of that.
+I see many developer not using raw CSS, but using preprocessors on top of that (with native CSS support this would look like that: `{ test: /\.sass$/, type: "stylesheet", use: "sass-loader" }`).
+
+As of [State of CSS 2020](https://2020.stateofcss.com/en-US/technologies/css-in-js/) CSS Modules is a popular way of writing modular CSS and it being a native module type in webpack allows to benefit from module graph optimization like Tree Shaking (Used Exports Optimization and Side-Effects optimization). When using CSS Modules this means the resulting CSS will only contains CSS rules that are referenced from the application (as one is used to from JS Tree Shaking).
+
+There are some potentical CSS Modules specific optimizations which are possible with the global knowledge of the application webpack has: CSS rules can be split into smaller rules to avoid repeating common properties. This can result in much smaller payload as the output CSS contains less repeated properties (Atomic CSS).
```

Comment:
```suggestion
There are some potential CSS Modules specific optimizations which are possible with the global knowledge of the application webpack has: CSS rules can be split into smaller rules to avoid repeating common properties. This can result in much smaller payload as the output CSS contains less repeated properties (Atomic CSS).
```